vimeo-networking/src/main/java/com/vimeo/networking/model/search/SearchResponse.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016 Vimeo
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.vimeo.networking.model.search;
+
+import com.google.gson.annotations.SerializedName;
+import com.vimeo.networking.model.BaseResponseList;
+
+/**
+ * Response of /search
+ * <p/>
+ * Created by zetterstromk on 6/27/16.
+ */
+public class SearchResponse extends BaseResponseList<SearchResult> {
+
+    private static final long serialVersionUID = -7915082057592438294L;
+
+    @SerializedName("facets")
+    public SearchFacetCollection mFacetCollection;
+
+    @Override
+    public Class getModelClass() {
+        return SearchResponse.class;
```

Comment:
This looks incorrect, in all other places we use BaseResponseList, getModelClass returns the same class as the parameterized type. So in this case it would return SearchResult.class. Could you also modify BaseResponseList so that getModelClass returns Class<T> so that we can catch this sort of error in the future via compiler error?